### PR TITLE
[Merged by Bors] - feat: port Algebra.Order.Field.Basic

### PR DESCRIPTION
**Pull request by mcdoll**

### CODE REVIEW

**Comment by mcdoll** [comment]
One error and one `and_comm'` warning left.

**Comment by Ruben-VandeVelde** [comment]
Fixed.

File: Mathlib/Algebra/Order/Field/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,987 @@
...
+
+open Function OrderDual
+
+variable {ι α β : Type _}
+
+section LinearOrderedSemifield
+
+variable [LinearOrderedSemifield α] {a b c d e : α} {m n : ℤ}
+
+/-- `equiv.mulLeft₀` as an order_iso. -/
```
Comment (semorrison) [inline Mathlib/Algebra/Order/Field/Basic.lean:29]:
```suggestion
/-- `Equiv.mulLeft₀` as an order_iso. -/
```

File: Mathlib/Algebra/Order/Field/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,987 @@
...
+@[simps (config := { simpRhs := true })]
+def OrderIso.mulLeft₀ (a : α) (ha : 0 < a) : α ≃o α :=
+  { Equiv.mulLeft₀ a ha.ne' with map_rel_iff' := @fun _ _ => mul_le_mul_left ha }
+#align order_iso.mul_left₀ OrderIso.mulLeft₀
+
+/-- `equiv.mulRight₀` as an order_iso. -/
```
Comment (semorrison) [inline Mathlib/Algebra/Order/Field/Basic.lean:35]:
```suggestion
/-- `Equiv.mulRight₀` as an order_iso. -/
```

File: Mathlib/Algebra/Order/Field/Basic.lean [hunk 3]
```diff
@@ -0,0 +1,987 @@
...
+-/
+
+
+/- TODO: Unify `add_halves` and `add_halves'` into a single lemma about
+`division_semiring` + `char_zero` -/
+theorem add_halves (a : α) : a / 2 + a / 2 = a := by
+  rw [div_add_div_same, ← two_mul, mul_div_cancel_left a two_ne_zero]
+#align add_halves add_halves
+
+-- TODO: Generalize to `division_semiring`
```
Comment (semorrison) [inline Mathlib/Algebra/Order/Field/Basic.lean:489]:
```suggestion
-- TODO: Generalize to `DivisionSemiring`
```

File: Mathlib/Algebra/Order/Field/Basic.lean [hunk 4]
```diff
@@ -0,0 +1,987 @@
...
+theorem IsLUB.mul_left {s : Set α} (ha : 0 ≤ a) (hs : IsLUB s b) :
+    IsLUB ((fun b => a * b) '' s) (a * b) := by
+  rcases lt_or_eq_of_le ha with (ha | rfl)
+  · exact (OrderIso.mulLeft₀ _ ha).isLUB_image'.2 hs
+  · simp_rw [zero_mul]
+    rw [hs.nonempty.image_const]
+    exact isLUB_singleton
+#align is_lub.mul_left IsLUB.mul_left
+
+-- TODO: Generalize to `LinearOrderedSemiield`
```
Comment (semorrison) [inline Mathlib/Algebra/Order/Field/Basic.lean:920]:
```suggestion
-- TODO: Generalize to `LinearOrderedSemifield`
```

File: Mathlib/Algebra/Order/Field/Basic.lean [hunk 5]
```diff
@@ -0,0 +1,987 @@
...
+  -- take inverses on both sides and use the assumption `2 ≤ a`.
+  convert (one_div a).le.trans (inv_le_inv_of_le zero_lt_two a2)
+  -- show `1 - 1 / 2 = 1 / 2`.
+  rw [sub_eq_iff_eq_add, ←two_mul, mul_inv_cancel two_ne_zero]
+#align sub_one_div_inv_le_two sub_one_div_inv_le_two
+
+/-! ### Results about `IsLUB` -/
+
+
+-- TODO: Generalize to `LinearOrderedSemiield`
```
Comment (semorrison) [inline Mathlib/Algebra/Order/Field/Basic.lean:910]:
```suggestion
-- TODO: Generalize to `LinearOrderedSemifield`
```

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**